### PR TITLE
Added AddTestQuestions API

### PR DESCRIPTION
**Pull request by Krishna92Soni**

### CODE REVIEW

File: APTITUDE/APTITUDETEST.Data/Data/TestRepository.cs [hunk 1]
```diff
@@ -165,6 +165,126 @@ public async Task<JsonResult> UpdateTestGroup(UpdateTestGroupVM updateTest)
...
+                }
+
+                #region Code to check questions are available in DB
+                Func<TestQuestionsCountVM, int> func = x => x.OneMarkQuestion;
+                var MarkQuestionCountReq = 0;
+                for (int questionType = 1; questionType <= 2; questionType++)
+                {
+                    for (int i = 1; i <= 5; i++)
+                    {
+                        int MarkQuestionCountInDB = await Task.FromResult(_context.Questions.Where(t => t.Topic == addTestQuestion.TopicId && t.QuestionType == questionType && t.Difficulty == i && t.IsDeleted == false).Count());
```
Comment (UrvishaNayee) [inline APTITUDE/APTITUDETEST.Data/Data/TestRepository.cs:192]: Get questions from db outside the loop, store in local variable and here get specific record from local variable

File: APTITUDE/APTITUDETEST.Data/Data/TestRepository.cs [hunk 2]
```diff
@@ -165,6 +165,126 @@ public async Task<JsonResult> UpdateTestGroup(UpdateTestGroupVM updateTest)
...
+                if (testQuestion != null)
+                {
+                    return new JsonResult(new ApiResponse<string>
+                    {
+                        Message = string.Format(ResponseMessages.TestTopicAlreadyExists),
+                        Result = true,
+                        StatusCode = ResponseStatusCode.OK
+                    });
+                }
+
```
Comment (UrvishaNayee) [inline APTITUDE/APTITUDETEST.Data/Data/TestRepository.cs:239]: include validation to match addTestQuestion.NoOfQuestion and total questions count added based on question type

File: APTITUDE/APTITUDETEST.Data/Data/TestRepository.cs (hunk 1 above)
Comment (Krishna92Soni) [reply APTITUDE/APTITUDETEST.Data/Data/TestRepository.cs]: Okay

File: APTITUDE/APTITUDETEST.Data/Common/ResponseMessages.cs [hunk 3]
```diff
@@ -29,6 +29,12 @@ public class ResponseMessages
         public static string TokenInvalid = "token is invalid";
         #endregion
 
+        #region Test
+        public static string NotEnoughQuestion = "There is not enough {0} mark questions in database";
+        public static string TestTopicAlreadyExists = "Questions for this topic already exists";
+        public static string NoOfQuestions = "Please enter correct No of Questions";
```
Comment (UrvishaNayee) [inline APTITUDE/APTITUDETEST.Data/Common/ResponseMessages.cs:35]: 'No of Questions' should be in lower case

**Review by UrvishaNayee — APPROVED** [review]

File: APTITUDE/APTITUDETEST.Data/Data/TestRepository.cs [hunk 4]
```diff
@@ -165,6 +166,137 @@ public async Task<JsonResult> UpdateTestGroup(UpdateTestGroupVM updateTest)
                 });
             }
 
+        }
+
+        public async Task<JsonResult> AddTestQuestions(AddTestQuestionsVM addTestQuestion)
```
Comment (mittalvyas1) [inline APTITUDE/APTITUDETEST.Data/Data/TestRepository.cs:171]: Mehod is too long. Create private method to shorten logic

**Pull request merged**